src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -66,9 +67,11 @@ struct MinimumSpanningTree
       Kokkos::Profiling::pushRegion("ArborX::MST::compute_core_distances");
       Kokkos::View<float *, MemorySpace> core_distances(
           "ArborX::MST::core_distances", n);
-      bvh.query(space, NearestK<Points>{points, k},
-                MaxDistance<Points, decltype(core_distances)>{points,
-                                                              core_distances});
+      bvh.query(
+          space,
+          Experimental::attach_indices(Experimental::nearest_k(points, k)),
```

Comment:
We used to attach `typename MemorySpace::size_type`.
Please confirm that you mean to change it to the current default that is IIRC `unsigned`?

Reply:
Yes. Imho, attaching `MemorySpace::size_type` is inconvenient, given that its signess and size change depending on the memory space.